skimmer/skimmer_utils.py

```diff
@@ -162,78 +161,43 @@ def get_phi_spike_filter(hot_spots_dict,var_name,j_eta_i,j_phi_i,rad):
-    elif channel == "s":
-        pass1 = get_phi_spike_filter(hot_spots_dict,"j1Phivsj1Eta",jetVar_i(jets_eta,0),jetVar_i(jets_phi,0),rad)
-        pass2 = get_phi_spike_filter(hot_spots_dict,"j2Phivsj2Eta",jetVar_i(jets_eta,1),jetVar_i(jets_phi,1),rad)
-    events = events[pass1 & pass2 & pass3 & pass4]
+    conditions = np.ones(len(events), dtype=bool)
+    for i in range(n_jets):
+        conditions &= __get_phi_spike_filter(hot_spots_dict,f"j{i+1}Phivsj{i+1}Eta",__jet_var_i(jets_eta,i),__jet_var_i(jets_phi,i),rad)
+    events = events[conditions]
     return events
 
 
-def apply_hem_veto(events):
-
-    if is_tree_maker(events):
-        jets = events.Jets
-        electrons = events.Electrons
-        muons = events.Muons
-
-    else:
-        jets = ak.zip({
-            "pt": events["Jet_pt"],
-            "eta": events["Jet_eta"],
-            "phi": events["Jet_phi"],
-        })
-        electrons = ak.zip({
-            "pt":  events["Electron_pt"],
-            "eta": events["Electron_eta"],
-            "phi": events["Electron_phi"],
-        })
-        muons = ak.zip({
-            "pt":  events["Muon_pt"],
-            "eta": events["Muon_eta"],
-            "phi": events["Muon_phi"],
-        })
-
-    jet_condition = obj.is_good_ak4_jet(jets)
-    electron_condition = obj.is_veto_electron(electrons)
-    muon_condition = obj.is_veto_muon(muons)
-    good_ak4_jets = jets[jet_condition]
-    veto_electrons = electrons[electron_condition]
-    veto_muons = muons[muon_condition]
-
+def apply_hem_veto(events, ak4_jets, electrons, muons):
     jet_hem_condition = (
-        (good_ak4_jets.eta > -3.05)
-        & (good_ak4_jets.eta < -1.35)
-        & (good_ak4_jets.phi > -1.62)
-        & (good_ak4_jets.phi < -0.82)
+        (ak4_jets.eta > -3.05)
+        & (ak4_jets.eta < -1.35)
+        & (ak4_jets.phi > -1.62)
+        & (ak4_jets.phi < -0.82)
     )
     electron_hem_condition = (
-        (veto_electrons.eta > -3.05)
-        & (veto_electrons.eta < -1.35)
-        & (veto_electrons.phi > -1.62)
-        & (veto_electrons.phi < -0.82)
+        (electrons.eta > -3.05)
+        & (electrons.eta < -1.35)
+        & (electrons.phi > -1.62)
+        & (electrons.phi < -0.82)
     )
     muon_hem_condition = (
-        (veto_muons.eta > -3.05)
-        & (veto_muons.eta < -1.35)
-        & (veto_muons.phi > -1.62)
-        & (veto_muons.phi < -0.82)
+        (muons.eta > -3.05)
+        & (muons.eta < -1.35)
+        & (muons.phi > -1.62)
+        & (muons.phi < -0.82)
     )
     veto = (
-        ((ak.num(jets) > 0) & ak.any(jet_hem_condition, axis=1))
-        | ((ak.num(muons) > 0) & ak.any(muon_hem_condition, axis=1))
-        | ((ak.num(electrons) > 0) & ak.any(electron_hem_condition, axis=1))
+        ((ak.num(ak4_jets) > 0) & ak.any(jet_hem_condition, axis=1))
+        | ((ak.num(electrons) > 0) & ak.any(muon_hem_condition, axis=1))
+        | ((ak.num(muons) > 0) & ak.any(electron_hem_condition, axis=1))
```

Comment:
Btw, I think we do not need the first condition on the number of objects btw:
```
>>> import awkward as ak
>>> x = ak.Array([[True, False], [], [False]])
>>> ak.any(x, axis=1)
<Array [True, False, False] type='3 * bool'>
>>> ~ak.any(x, axis=1)
<Array [False, True, True] type='3 * bool'>
>>> y = ak.Array([[1, 2], [], [3]])
>>> y[~ak.any(x, axis=1)]
<Array [[], [3]] type='2 * var * int64'>
```
But it's fine to keep it like this for now, I will test in the future on one dataset with and without to check if it gives the same :)
I just want to remove it in case it is useless in order to avoid typos like here ;) Fewer lines of code, fewer bugs ;)

Reply:
Okay, it looks like from your example snippet the first condition is not necessary. I will just go ahead and remove it now.

Reply:
My coffea code has the same conditions as above. I tried removing the conditions on number of physical objects, and it gave the same results.

Reply:
Perfect!